backend/src/app.ts

```diff
@@ -0,0 +1,255 @@
+
+  const modulesRestApis: ModuleRestApi[] = modules.map((module) => module.restApi).filter(isDefined);
+  const restApi = restApiExpressServer(modulesRestApis);
+
+  await initializeDummyData(commandBus, entityIdGenerator);
+
+  return { restApi };
+}
+
+//TODO: Remove for production usage
+async function initializeDummyData(commandBus: CommandBus, entityIdGenerator: EntityIdGenerator) {
+  const janKowalski = {
+    playerId: entityIdGenerator.generate(),
+    firstName: 'Jan',
+    emailAddress: 'jan.kowalski@test.pl',
+    lastName: 'Kowalski',
+    phoneNumber: '123321333',
+  };
+  const katarzynaNowak = {
+    playerId: entityIdGenerator.generate(),
+    firstName: 'Katarzyna',
+    emailAddress: 'kasia12@test.pl',
+    lastName: 'Nowak',
+    phoneNumber: '143351333',
+  };
+  const tomekDomek = {
+    playerId: entityIdGenerator.generate(),
+    firstName: 'Tomek',
+    emailAddress: 'tomek.domek@test.pl',
+    lastName: 'Domek',
+    phoneNumber: '123321335',
+  };
+  const franekPoranek = {
+    playerId: entityIdGenerator.generate(),
+    firstName: 'Franek',
+    emailAddress: 'franek.ranek@test.pl',
+    lastName: 'Ranek',
+    phoneNumber: '123321334',
+  };
+
+  await commandBus.execute(new CreatePlayerProfile({ ...janKowalski }));
+  await commandBus.execute(new CreatePlayerProfile({ ...katarzynaNowak }));
+  await commandBus.execute(new CreatePlayerProfile({ ...tomekDomek }));
+  await commandBus.execute(new CreatePlayerProfile({ ...franekPoranek }));
+}
+
+function TournamentRegistrationsRepository() {
+  if (process.env.MONGO_REPOSITORIES === 'ENABLED' && process.env.TOURNAMENTS_REGISTRATIONS_DATABASE === 'MONGO') {
+    return new MongoTournamentRegistrationsRepository();
+  }
+  if (process.env.POSTGRES_REPOSITORIES === 'ENABLED' && process.env.TOURNAMENTS_REGISTRATIONS_DATABASE === 'POSTGRES') {
+    return new PostgreSqlTournamentRegistrationsRepository();
+  }
+  return new InMemoryTournamentRegistrationsRepository();
+}
+
+function PlayerProfilesRepository() {
+  if (process.env.MONGO_REPOSITORIES === 'ENABLED' && process.env.TOURNAMENTS_REGISTRATIONS_DATABASE === 'MONGO') {
+    return new MongoPlayerProfileRepository();
+  }
+  return new InMemoryPlayerProfileRepository();
+}
+
+function DoublesTournamentRepository() {
+  if (process.env.MONGO_REPOSITORIES === 'ENABLED' && process.env.DOUBLES_TOURNAMENT_DATABASE === 'MONGO') {
+    return new MongoDoublesTournamentRepository();
+  }
+  return new InMemoryDoublesTournamentRepository();
+}
+
+function MatchRepository() {
+  if (process.env.MONGO_REPOSITORIES === 'ENABLED' && process.env.MATCH_DATABASE === 'MONGO') {
+    return new MongoMatchRepository();
+  }
+  return new InMemoryMatchRepository();
+}
+
+function TournamentTablesRepository() {
+  if (process.env.MONGO_REPOSITORIES === 'ENABLED' && process.env.MATCH_DATABASE === 'MONGO') {
+    return new MongoTournamentTablesRepository();
+  }
+  return new InMemoryTournamentTablesRepository();
+}
+
```

Comment:
Powtarzacie wiele razy ten sam warunek aby pobrać repozytorium dla bazy. Fajnie byłoby zastąpić dwiema klasami z metodami zwracającymi te repozytoria, i jedną funkcje, która w zależności od warunku zwróci odpowiednią klasę. Raz, że warunek jest jeden, to np. zmiana bazy danych wymaga znacznie mniej pracy zwłaszcza jak ilość tych repozytoriów się rozrasta.